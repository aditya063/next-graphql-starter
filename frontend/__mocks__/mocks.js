export const user = {
  id: '1',
  email: 'a@a.com'
};

export const idea = {
  id: '1',
  content: 'mock content'
};

export const newIdea = {
  id: '1',
  content: 'new mock content'
};

export const users = [user];
export const ideas = [user];
