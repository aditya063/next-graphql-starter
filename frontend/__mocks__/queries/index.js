export {
  MOCK_IS_LOGGED_IN,
  MOCK_CURRENT_USER,
  MOCK_ERROR_CURRENT_USER,
  MOCK_SIGN_UP,
  MOCK_SIGN_IN,
  MOCK_SIGN_OUT,
  MOCK_REQUEST_RESET,
  MOCK_RESET_PASSWORD
} from './user';

export {
  MOCK_CREATE_IDEA,
  MOCK_UPDATE_IDEA,
  MOCK_DELETE_IDEA
} from './idea';
