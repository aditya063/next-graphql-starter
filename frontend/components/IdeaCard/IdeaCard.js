import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import * as query from './IdeaCard.query';
import * as sc from './IdeaCard.style';

class IdeaCard extends React.PureComponent {
  state = { content: this.props.content };

  handleChangeideaInput = (e, updateIdea) => {
    this.setState({ content: e.target.value }, updateIdea);
  };

  handleClickDeleteBtn = (e, deleteIdea) => {
    e.target.disabled = true;
    deleteIdea();
  };

  handleError = error => error;

  render() {
    return (
      <sc.li>
        <Mutation
          mutation={query.DELETE_IDEA_MUTATION}
          variables={{ id: this.props.id }}
          refetchQueries={[{ query: query.ME_IDEAS_QUERY }]}
          onError={this.handleError}
          errorPolicy="all"
        >
          {deleteIdea => (
            <sc.deleteBtn
              type="button"
              onClick={e => this.handleClickDeleteBtn(e, deleteIdea)}
            />
          )}
        </Mutation>

        <Mutation
          mutation={query.UPDATE_IDEA_MUTATION}
          variables={{
            id: this.props.id,
            content: this.state.content
          }}
          onError={this.handleError}
          errorPolicy="all"
        >
          {updateIdea => (
            <sc.ideaInput
              type="text"
              value={this.state.content}
              onChange={e => this.handleChangeideaInput(e, updateIdea)}
            />
          )}
        </Mutation>
      </sc.li>
    );
  }
}

IdeaCard.propTypes = {
  id: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
};

export default IdeaCard;
