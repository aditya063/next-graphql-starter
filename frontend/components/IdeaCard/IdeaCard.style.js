import styled from 'styled-components';
import { List, Input, Icon } from 'antd';

export const IdeaCard = styled(List.Item)`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
`;

export const IdeaInput = styled(Input.TextArea).attrs({
  autosize: { minRows: 1, maxRows: 10 }
})``;

export const DetailIcon = styled(Icon).attrs({
  type: 'up-square',
  theme: 'twoTone'
})`
  font-size: 1.2rem;
`;

export const DeleteIcon = styled(Icon).attrs({
  type: 'close-square',
  theme: 'twoTone'
})`
  font-size: 1.2rem;
`;
