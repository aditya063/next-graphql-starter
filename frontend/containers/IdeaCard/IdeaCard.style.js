import styled from 'styled-components';
import { List } from 'antd';

export const IdeaCard = styled(List.Item)`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
`;
