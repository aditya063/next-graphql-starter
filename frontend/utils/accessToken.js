import fetch from 'isomorphic-unfetch';
// import cookie from 'cookie';
// import jwt from 'jsonwebtoken';

let accessToken = '';

export const setAccessToken = token => (accessToken = token);

export const getAccessToken = () => accessToken;

export const clearAccessToken = () => (accessToken = '');

export const fetchAccessToken = async refreshToken => {
  const url =
    process.env.NODE_ENV === 'development'
      ? process.env.DEV_REFRESH_URL
      : process.env.PROD_REFRESH_URL;

  const options = {
    method: 'GET',
    credentials: 'include',
    headers: { cookie: `rt=${refreshToken}` }
  };

  try {
    const response = await fetch(url, options);

    const data = await response.json();

    return data.accessToken;
  } catch (error) {
    if (process.env.NODE_ENV === 'development')
      console.error('fetchAccessToken error: ', error);
  }
};

// export const verifyAccessToken = accessToken => {
//   try {
//     jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET);
//   } catch (error) {
//     if (process.env.NODE_ENV === 'development')
//       console.error('fetchAccessToken error: ', error);
//   }
// };

// const checkAccessToken = async cookies => {
//   const accessToken = getAccessToken();

//   if (!accessToken) return true;

//   const parsedCookies = cookie.parse(cookies);

//   if (!parsedCookies.rt) return '';

//   // const response = await fetchAccessToken(parsedCookies.rt);

//   // const data = await response.json();

//   // setAccessToken(data.accessToken);
//   // setAccessTokenExpiry(data.exp);

//   // return data.accessToken;
// };
