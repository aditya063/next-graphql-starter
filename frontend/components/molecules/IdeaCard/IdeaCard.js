import PropTypes from 'prop-types';
import DetailIcon from '../DetailIcon/DetailIcon';
import DeleteIcon from '../DeleteIcon/DeleteIcon';
import IdeaInput from '../IdeaInput/IdeaInput';
import * as sc from './IdeaCard.style';

const IdeaCard = props => (
  <sc.IdeaCard data-testid='IdeaCard'>
    <DetailIcon id={props.id} />

    <DeleteIcon id={props.id} />

    <IdeaInput id={props.id} content={props.content} />
  </sc.IdeaCard>
);

IdeaCard.propTypes = {
  id: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
};

export default React.memo(IdeaCard);
