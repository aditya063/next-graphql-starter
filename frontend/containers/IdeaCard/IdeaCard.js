import PropTypes from 'prop-types';

import DeleteIcon from '../../components/DeleteIcon/DeleteIcon';
import * as sc from './IdeaCard.style';
import IdeaInput from '../../components/IdeaInput.js/IdeaInput';
import DetailIcon from '../../components/DetailIcon/DetailIcon';

const IdeaCard = React.memo(props => (
  <sc.IdeaCard>
    <DetailIcon id={props.id} />

    <DeleteIcon id={props.id} />

    <IdeaInput id={props.id} content={props.content} />
  </sc.IdeaCard>
));

IdeaCard.propTypes = {
  id: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
};

export default IdeaCard;
