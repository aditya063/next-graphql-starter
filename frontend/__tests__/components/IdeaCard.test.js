/* eslint-disable no-console */
import { mount } from 'enzyme';
import { ApolloConsumer } from 'react-apollo';
import { MockedProvider } from 'react-apollo/test-utils';
import { load } from '../../utils/load';
import IdeaCard from '../../components/IdeaCard/IdeaCard';
import {
  MOCK_UPDATE_IDEA_MUTATION,
  MOCK_DELETE_IDEA_MUTATION
} from '../../components/IdeaCard/IdeaCard.query';
import {
  ME_IDEAS_QUERY,
  MOCK_ME_IDEAS_QUERY
} from '../../components/IdeaCardContainer/IdeaCardContainer.query';
import * as mock from '../../__mocks__/mocks';

describe('IdeaCard', () => {
  let mockProps;
  let mockQueries;
  let apolloClient;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = { id: '1', content: 'mock idea 1 content' };
    mockQueries = [
      MOCK_UPDATE_IDEA_MUTATION,
      MOCK_DELETE_IDEA_MUTATION,
      MOCK_ME_IDEAS_QUERY
    ];

    wrapper = mount(
      // <MockedProvider mocks={mockQueries} addTypename={false}>
      //   <IdeaCard {...mockProps} />
      // </MockedProvider>,
      <ApolloConsumer>
        {client => {
          apolloClient = client;
          return <IdeaCard {...mockProps} />;
        }}
      </ApolloConsumer>,
      {
        disableLifecycleMethods: true,
        wrappingComponent: MockedProvider,
        wrappingComponentProps: { mocks: mockQueries, addTypename: false }
      }
    );
  });

  it('matches snapshot', () => {
    // expect(
    //   wrapper
    //     .find('Mutation')
    //     .at(0)
    //     .renderProp('children')()
    // ).toMatchSnapshot();

    // expect(
    //   wrapper
    //     .find('Mutation')
    //     .at(1)
    //     .renderProp('children')()
    // ).toMatchSnapshot();

    expect(wrapper).toMatchSnapshot();
  });

  fit('matches snapshot - update idea - loading', async () => {
    console.log(wrapper.debug());

    // const handleClickDeleteBtn = (wrapper.find('IdeaCard').instance().handleClickDeleteBtn() = jest.fn());

    // const userIdeas1 = await apolloClient.query({ query: ME_IDEAS_QUERY });
    // console.log(': userIdeas1', userIdeas1.data.getUserIdeas);

    const ideaInput1 = wrapper.find('IdeaCardstyle__deleteBtn');
    // console.log(': ideaInput1', ideaInput1.debug());

    // await load(wrapper);

    // console.log(ideaInput1.props());

    ideaInput1.prop('onClick')({ target: { disabled: false } });

    console.log(wrapper.debug());

    // expect(
    //   wrapper.find('IdeaCard').instance().handleClickDeleteBtn
    // ).toHaveBeenCalledTimes(1);

    // await load(wrapper);

    // console.log(wrapper.find('IdeaCard').instance().state);

    // console.log(apolloClient.readQuery({ query: ME_IDEAS_QUERY }));

    // const userIdeas2 = await apolloClient.query({ query: ME_IDEAS_QUERY });
    // console.log(': userIdeas2', userIdeas2);

    // const ideaInput2 = wrapper.find('input');
    // console.log(': ideaInput2', ideaInput2.debug());

    // // wrapper.update();

    // console.log(wrapper.debug());
  });
});
