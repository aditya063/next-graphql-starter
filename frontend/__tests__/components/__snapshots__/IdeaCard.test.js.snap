// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard matches snapshot 1`] = `
<IdeaCardstyle__li
  snapshot="IdeaCard"
>
  <Mutation
    errorPolicy="all"
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "DELETE_IDEA_MUTATION",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "deleteIdea",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 90,
          "start": 0,
        },
      }
    }
    onError={[Function]}
    refetchQueries={
      Array [
        Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "ME_IDEAS_QUERY",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "getUserIdeas",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "content",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 78,
              "start": 0,
            },
          },
        },
      ]
    }
    variables={
      Object {
        "id": undefined,
      }
    }
  >
    <Component />
  </Mutation>
  <Mutation
    errorPolicy="all"
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "UPDATE_IDEA_MUTATION",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "content",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "updateIdea",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "content",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 142,
          "start": 0,
        },
      }
    }
    onError={[Function]}
    refetchQueries={
      Array [
        Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "ME_IDEAS_QUERY",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "getUserIdeas",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "content",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 78,
              "start": 0,
            },
          },
        },
      ]
    }
    variables={
      Object {
        "content": undefined,
        "id": undefined,
      }
    }
  >
    <Component />
  </Mutation>
</IdeaCardstyle__li>
`;
